### PR TITLE
Allow changing the Houdini generated output directory

### LINKED ISSUE #391: Changing the houdini output folder
I'm currently reading https://www.houdinigraphql.com/api/config and I can't seem to find the field that changes the path or name of the ``$houdini`` folder.

### PR DESCRIPTION
**Pull request by endigma**
Fixes #391 

# Configurable output directory

This PR adds a config value to configure the output directory name for Houdini's generated files (`$houdini`) by default.

## TODO

- [x] fixup `cmd/init` to work properly
- [x] somehow drag config into `is_root_route` in `houdini-svelte/fsPatch`
- [x] update `e2e` to use `.houdini`
- [x] changset?
- [x] docs update

### CODE REVIEW

**Comment by SeppahBaws** [comment]
uh-oh, can you add `.houdini/` to the gitignore? 😅

**Comment by endigma** [comment]
whoops

**Comment by endigma** [comment]
Running `pnpm test` on this branch and main results in the same 2 tests failing, and E2E is failing in GHA on build but this also succeeds locally, any insight?

**Comment by endigma** [comment]
https://github.com/sveltejs/kit/issues/13002 blocks fixing the failing thing _properly_ but I can just switch the e2e to use the barrel exports. Let me know what you want done re: getting this done, if you want to merge the functionality ASAP I see a few options:

- Disable the client option and merge the plumbing with `$houdini` as the default
- Merge as-is, making `.houdini` the new default; add a note to docs about using the barreled exports.
- Merge as-is, but leaving `$houdini` as the default for new projects
- Wait until the SvelteKit bug is addressed to proceed (likely gonna hit a huge conflict when it comes time to resurrect unless I go and fix this upstream myself ASAP)

File: packages/houdini-svelte/src/plugin/fsPatch.ts [hunk 1]
```diff
@@ -280,7 +280,7 @@ function is_root_route(filepath: PathLike): boolean {
 		filepath.endsWith(path.join('src', 'routes')) &&
 		// ignore the src/routes that exists in the type roots
 		!filepath.includes('.svelte-kit') &&
-		!filepath.includes('$houdini')
+		!filepath.includes(_config.runtimeDir ?? '$houdini')
```
Comment (SeppahBaws) [inline packages/houdini-svelte/src/plugin/fsPatch.ts:283]: Shouldn't this default to '.houdini' if the runtimeDir hasn't been set?

File: packages/houdini/src/cmd/init.ts
Comment (SeppahBaws) [inline packages/houdini/src/cmd/init.ts:1]: Do the TODOs in this file still apply, or can they be removed?

**Review by SeppahBaws — COMMENTED** [review]
We should also update the documentation site to reflect the updated config structure.

File: packages/houdini-svelte/src/plugin/fsPatch.ts (hunk 1 above)
Comment (endigma) [reply packages/houdini-svelte/src/plugin/fsPatch.ts:283]: The default for `init` and some other things is now `.houdini`, but the hardcoded default has to remain `$houdini` to preserve backward compatibility afaik

**Comment by endigma** [comment]
Tests should pass now that I've fixed the path resolving bug, and I've updated docs to the best of my knowledge, please let me know if I've done it wrong or if that file is somehow autogenerated.

File: packages/houdini-svelte/src/plugin/fsPatch.ts (hunk 1 above)
Comment (AlecAivazis) [reply packages/houdini-svelte/src/plugin/fsPatch.ts:283]: can you add a comment somewhere to remind us that this needs to match the default value?

Comment (endigma) [reply packages/houdini-svelte/src/plugin/fsPatch.ts:285]: Done

**Review by AlecAivazis — APPROVED** [review]

**Pull request merged**